pkgs/os-specific/darwin/apple-sdk-11.0/cf-setup-hook.sh

```diff
@@ -1,6 +1,14 @@
 forceLinkCoreFoundationFramework() {
-  NIX_CFLAGS_COMPILE="-F@out@/Library/Frameworks${NIX_CFLAGS_COMPILE:+ }${NIX_CFLAGS_COMPILE-}"
-  NIX_LDFLAGS+=" @out@/Library/Frameworks/CoreFoundation.framework/CoreFoundation.tbd"
+  local role_post
+  getHostRole
+
+  local cflags_compile_role_var="NIX_CFLAGS_COMPILE${role_post}"
+  local cflags_compile_role="${!cflags_compile_role_var-}"
+  export NIX_CFLAGS_COMPILE${role_post}="-F@out@/Library/Frameworks${cflags_compile_role:+ $cflags_compile_role}"
+
+  local ldflags_role_var="NIX_LDFLAGS${role_post}"
+  local ldflags_role="${!ldflags_role_var-}"
+  export NIX_LDFLAGS${role_post}+="${ldflags_role:+ }@out@/Library/Frameworks/CoreFoundation.framework/CoreFoundation.tbd"
 }
 
-preConfigureHooks+=(forceLinkCoreFoundationFramework)
```

Comment:
You mentioned somewhere this change was necessary to keep `role_post` in scope?

When do hooks run without such assignment and might we want to just pass the value in to keep it in scope? `preConfigureHooks+=(forceLinkCoreFoundationFramework getHostRole)` or something?

Reply:
I'm not sure if that's actually possible. In the snippet above "forceLinkCoreFoundationFramework" and "getHostRole" are literal strings appended to an array.

Though I can't think of any reason why `forceLinkCoreFoundationFramework` would need to run specifically in the `configure` phase. All it does is append compiler/linker flags. These flags can be set at any point before the build phase. They could even be set in the `.drv` itself.